wallet-test/src/test/scala/org/bitcoins/wallet/TrezorAddressTest.scala

```diff
@@ -0,0 +1,266 @@
+      case other                   => fail(s"unknown purpose: $other")
+    }
+
+    for {
+      wallet <- getWallet(conf)
+      existingAccounts <- wallet.listAccounts(purpose)
+      _ <- {
+        val accountsToCreate = existingAccounts.length until vectors.length
+
+        FutureUtil.sequentially(accountsToCreate) { _ =>
+          wallet.createNewAccount(purpose)
+        }
+      }
+      accounts <- wallet.listAccounts(purpose)
+
+      // we want to find all accounts for the given account type,
+      // and match it with its corresponding test vector
+      accountsWithVectors = {
+        assert(accounts.length == vectors.length)
+        val accountsWithVectors = vectors.map { vec =>
+          accounts.find(_.hdAccount.index == vec.account) match {
+            case None =>
+              fail(
+                s"Did not find account in wallet with index ${vec.account}. Accounts: ${accounts.mkString}")
+            case Some(account) =>
+              assert(account.xpub == vec.xpub)
+              account -> vec
+          }
+        }
+        accountsWithVectors
+      }
+
+      // here we generate addresses matching the ones found
+      // in the accompanying test vector for each account
+      // at the end we group them all together
+      accountsWithAddrsWithVecs <- {
+        FutureUtil.sequentially(accountsWithVectors) {
+          case (acc, vec) =>
+            val addrFutures: Future[Seq[AddressDb]] =
+              FutureUtil.sequentially(vec.addresses) { vector =>
+                val addrFut = vector.chain match {
+                  case Change => wallet.getNewChangeAddress(acc)
+                  case External =>
+                    wallet.getNewAddress(acc)
+                }
+                addrFut.flatMap(wallet.addressDAO.findAddress).map {
+                  case Some(addr) => addr
+                  case None =>
+                    fail(s"Did not find address we just generated in DAO!")
+                }
+              }
+            addrFutures.map(AccountAndAddrsAndVector(acc, _, vec))
+        }
+      }
+    } yield {
+      // lastly we loop over all accounts, addresses and vectors
+      // and verify that they are all the same
+      accountsWithAddrsWithVecs.foreach {
+        case AccountAndAddrsAndVector(account, addresses, vec) =>
+          val acctIdx = account.hdAccount.index
+          val vec = vectors.find(_.xpub == account.xpub) match {
+            case None =>
+              fail(s"Did not find test vector for account $acctIdx")
+            case Some(v) => v
+          }
+          assert(vec.addresses.length == addresses.length)
+
+          val sortedAddresses = addresses.sortBy(_.path.toString)
+          val sortedVectors = vec.addresses.sortBy(_.path.toString)
+          sortedAddresses.zip(sortedVectors).foreach {
+            case (foundAddress, expectedAddress) =>
+              assert(foundAddress.address == expectedAddress.address)
+          }
+      }
+      succeed
```

Comment:
I don't really like passing in `suceed` at the end, is there no way to actually get a `Future[Vector[Assertion]]` or something like that? @rorp pointed out yesterday we tests that were trivially passing in the eclair project because we were doing this succeed at the end which is obviously very bad.